### PR TITLE
Charmhub: Optimise channel map find and subset filtering.

### PR DESCRIPTION
**Pull request by SimonRichardson**
The following ensures that we're only requesting the fields that we
require when using info and find. Unfortunately this doesn't know about
channel-map metadata.yaml and config.yaml so that needs to be worked in
correctly.

There is a missing piece of work where we'll need to request the info for a given
channel that will pass back the metadata.yaml and config.yaml back. This is
found in `apiserver/facades/client/charms/reqpositories.go#L169`. This is
where we look up the info channel map by either the channel or the revision.

Once the new API is up to do that, we should change this to correctly implement
that lookup.

## QA steps

What currently works:

```sh
juju bootstrap lxd test
juju info ubuntu
juju find ubuntu
juju find
```

### CODE REVIEW

File: charmhub/filter.go [hunk 1]
```diff
@@ -34,17 +34,20 @@ var defaultResultFilter = []string{
...
-	"result.media.url",
-	"result.media.width",
 	"result.publisher.display-name",
 	"result.store-url",
 	"result.summary",
 	"result.used-by",
 	"result.website",
 }
 
+var defaultMediaFilter = []string{
```
Comment (hmlanigan) [inline charmhub/filter.go:44]: I think we can drop the result.media in general.

File: charmhub/find.go [hunk 2]
```diff
@@ -39,7 +39,14 @@ func (c *FindClient) Find(ctx context.Context, query string) ([]transport.FindRe
 		return nil, errors.Trace(err)
 	}
 
-	path, err = path.Query("fields", defaultFindFilter())
+	var fields string
+	if query != "" {
+		fields = defaultFindFilter()
+	} else {
+		fields = subsetFindFilter()
+	}
```
Comment (hmlanigan) [inline charmhub/find.go:47]: Let's go for it and only request the Find data we use: names, bundle, version, publisher and summary?  We do not print anything different based on query vs all.

**Review by hmlanigan — APPROVED** [review]
QA is good.

**Comment by SimonRichardson** [comment]
$$merge$$

**Comment by SimonRichardson** [comment]
$$merge$$

> FAIL: txnwatcher_test.go:281: TxnWatcherSuite.TestInsertThenRemove

**Pull request merged**